src/main/java/com/redhat/pantheon/servlet/AsciidocRenderingServlet.java

```diff
@@ -145,9 +149,11 @@ private Content generateHtml(SlingHttpServletRequest request, Resource resource)
         }
 
         long start = System.currentTimeMillis();
-        c.html = getAsciidoctorInstance(resource).convert(
+        Asciidoctor asciidoctor = asciidoctorPoolService.requestInstance(resource);
+        c.html = asciidoctor.convert(
                 c.asciidoc,
                 ob.get());
+        asciidoctorPoolService.releaseInstance(asciidoctor);
```

Comment:
I think this should be in a `try - finally` block, to guarantee the instance gets returned even in the case of runtime exceptions.

Reply:
Good idea!